base58/__init__.py

```diff
@@ -53,16 +54,20 @@ def b58encode(
     """
     v = scrub_input(v)
 
-    nPad = len(v)
+    origlen = len(v)
     v = v.lstrip(b'\0')
-    nPad -= len(v)
+    newlen = len(v)
+
+    acc = int.from_bytes(v, byteorder='big')  # first byte is most significant
 
-    p, acc = 1, 0
-    for c in reversed(v):
-        acc += p * c
-        p = p << 8
     result = b58encode_int(acc, default_one=False, alphabet=alphabet)
-    return alphabet[0:1] * nPad + result
+    return alphabet[0:1] * (origlen - newlen) + result
+
+
+@lru_cache
```

Comment:
looks like mypy gets a bit confused by this even though it's correct, changing it to  `@lru_cache()` seems to resolve the issue